File: sdk/csharp/src/IO.Swagger/Model/RealTimeInformation.cs
/* 
 * RTPI REST Web Services API
 *
 * <p>RTPI REST Web Services API (<a href=\"https://goo.gl/s6NTJe\">RRWS API</a>) provides REST interface to retrieve information on real time bus information, timetables and bus stops. </p> <p>Each method available in RRWS API will be invoked using plain HTTP Request (GET method). Depending on parameters provided, result will be returned either in XML or JSON format (if format is not provided then JSON is used by default). Response will contain requested information (see method specification for details) and error status of the request.</p>  <h2>Error Codes</h2> <table> <tr><th>Error Code</th><th>Description</th></tr> <tr><td>0</td><td>Success</td></tr> <tr><td>1</td><td>No Results</td></tr> <tr><td>2</td><td>Missing Parameter</td></tr> <tr><td>3</td><td>Invalid Parameter</td></tr> <tr><td>4</td><td>Sceduled Downtime</td></tr> <tr><td>5</td><td>Unexpected System Error</td></tr> </table> <p>Error code is included in response for each message. For some error codes additional details will be provided in free text format (Error Message field).</p> 
 *
 * OpenAPI spec version: 1.0.0
 * 
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// RealTimeInformation
    /// </summary>
    [DataContract]
    public partial class RealTimeInformation :  IEquatable<RealTimeInformation>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="RealTimeInformation" /> class.
        /// </summary>
        /// <param name="Arrivaldatetime">Arrivaldatetime.</param>
        /// <param name="Duetime">Duetime.</param>
        /// <param name="Departuredatetime">Departuredatetime.</param>
        /// <param name="Departureduetime">Departureduetime.</param>
        /// <param name="Scheduledarrivaldatetime">Scheduledarrivaldatetime.</param>
        /// <param name="Scheduleddeparturedatetime">Scheduleddeparturedatetime.</param>
        /// <param name="Destination">Destination.</param>
        /// <param name="Destinationlocalized">Destinationlocalized.</param>
        /// <param name="Origin">Origin.</param>
        /// <param name="Originlocalized">Originlocalized.</param>
        /// <param name="Direction">Direction.</param>
        /// <param name="_Operator">_Operator.</param>
        /// <param name="Additionalinformation">Additionalinformation.</param>
        /// <param name="Lowfloorstatus">Lowfloorstatus.</param>
        /// <param name="Route">Route.</param>
        /// <param name="Sourcetimestamp">Sourcetimestamp.</param>
        /// <param name="Monitored">Monitored.</param>
        public RealTimeInformation(string Arrivaldatetime = null, string Duetime = null, string Departuredatetime = null, string Departureduetime = null, string Scheduledarrivaldatetime = null, string Scheduleddeparturedatetime = null, string Destination = null, string Destinationlocalized = null, string Origin = null, string Originlocalized = null, string Direction = null, string _Operator = null, string Additionalinformation = null, string Lowfloorstatus = null, string Route = null, string Sourcetimestamp = null, string Monitored = null)
        {
            this.Arrivaldatetime = Arrivaldatetime;
            this.Duetime = Duetime;
            this.Departuredatetime = Departuredatetime;
            this.Departureduetime = Departureduetime;
            this.Scheduledarrivaldatetime = Scheduledarrivaldatetime;
            this.Scheduleddeparturedatetime = Scheduleddeparturedatetime;
            this.Destination = Destination;
            this.Destinationlocalized = Destinationlocalized;
            this.Origin = Origin;
            this.Originlocalized = Originlocalized;
            this.Direction = Direction;
            this._Operator = _Operator;
            this.Additionalinformation = Additionalinformation;
            this.Lowfloorstatus = Lowfloorstatus;
            this.Route = Route;
            this.Sourcetimestamp = Sourcetimestamp;
            this.Monitored = Monitored;
        }
        
        /// <summary>
        /// Gets or Sets Arrivaldatetime
        /// </summary>
        [DataMember(Name="arrivaldatetime", EmitDefaultValue=false)]
        public string Arrivaldatetime { get; set; }
        /// <summary>
        /// Gets or Sets Duetime
        /// </summary>
        [DataMember(Name="duetime", EmitDefaultValue=false)]
        public string Duetime { get; set; }
        /// <summary>
        /// Gets or Sets Departuredatetime
        /// </summary>
        [DataMember(Name="departuredatetime", EmitDefaultValue=false)]
        public string Departuredatetime { get; set; }
        /// <summary>
        /// Gets or Sets Departureduetime
        /// </summary>
        [DataMember(Name="departureduetime", EmitDefaultValue=false)]
        public string Departureduetime { get; set; }
        /// <summary>
        /// Gets or Sets Scheduledarrivaldatetime
        /// </summary>
        [DataMember(Name="scheduledarrivaldatetime", EmitDefaultValue=false)]
        public string Scheduledarrivaldatetime { get; set; }
        /// <summary>
        /// Gets or Sets Scheduleddeparturedatetime
        /// </summary>
        [DataMember(Name="scheduleddeparturedatetime", EmitDefaultValue=false)]
        public string Scheduleddeparturedatetime { get; set; }
        /// <summary>
        /// Gets or Sets Destination
        /// </summary>
        [DataMember(Name="destination", EmitDefaultValue=false)]
        public string Destination { get; set; }
        /// <summary>
        /// Gets or Sets Destinationlocalized
        /// </summary>
        [DataMember(Name="destinationlocalized", EmitDefaultValue=false)]
        public string Destinationlocalized { get; set; }
        /// <summary>
        /// Gets or Sets Origin
        /// </summary>
        [DataMember(Name="origin", EmitDefaultValue=false)]
        public string Origin { get; set; }
        /// <summary>
        /// Gets or Sets Originlocalized
        /// </summary>
        [DataMember(Name="originlocalized", EmitDefaultValue=false)]
        public string Originlocalized { get; set; }
        /// <summary>
        /// Gets or Sets Direction
        /// </summary>
        [DataMember(Name="direction", EmitDefaultValue=false)]
        public string Direction { get; set; }
        /// <summary>
        /// Gets or Sets _Operator
        /// </summary>
        [DataMember(Name="operator", EmitDefaultValue=false)]
        public string _Operator { get; set; }
        /// <summary>
        /// Gets or Sets Additionalinformation
        /// </summary>
        [DataMember(Name="additionalinformation", EmitDefaultValue=false)]
        public string Additionalinformation { get; set; }
        /// <summary>
        /// Gets or Sets Lowfloorstatus
        /// </summary>
        [DataMember(Name="lowfloorstatus", EmitDefaultValue=false)]
        public string Lowfloorstatus { get; set; }
        /// <summary>
        /// Gets or Sets Route
        /// </summary>
        [DataMember(Name="route", EmitDefaultValue=false)]
        public string Route { get; set; }
        /// <summary>
        /// Gets or Sets Sourcetimestamp
        /// </summary>
        [DataMember(Name="sourcetimestamp", EmitDefaultValue=false)]
        public string Sourcetimestamp { get; set; }
        /// <summary>
        /// Gets or Sets Monitored
        /// </summary>
        [DataMember(Name="monitored", EmitDefaultValue=false)]
        public string Monitored { get; set; }
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class RealTimeInformation {\n");
            sb.Append("  Arrivaldatetime: ").Append(Arrivaldatetime).Append("\n");
            sb.Append("  Duetime: ").Append(Duetime).Append("\n");
            sb.Append("  Departuredatetime: ").Append(Departuredatetime).Append("\n");
            sb.Append("  Departureduetime: ").Append(Departureduetime).Append("\n");
            sb.Append("  Scheduledarrivaldatetime: ").Append(Scheduledarrivaldatetime).Append("\n");
            sb.Append("  Scheduleddeparturedatetime: ").Append(Scheduleddeparturedatetime).Append("\n");
            sb.Append("  Destination: ").Append(Destination).Append("\n");
            sb.Append("  Destinationlocalized: ").Append(Destinationlocalized).Append("\n");
            sb.Append("  Origin: ").Append(Origin).Append("\n");
            sb.Append("  Originlocalized: ").Append(Originlocalized).Append("\n");
            sb.Append("  Direction: ").Append(Direction).Append("\n");
            sb.Append("  _Operator: ").Append(_Operator).Append("\n");
            sb.Append("  Additionalinformation: ").Append(Additionalinformation).Append("\n");
            sb.Append("  Lowfloorstatus: ").Append(Lowfloorstatus).Append("\n");
            sb.Append("  Route: ").Append(Route).Append("\n");
            sb.Append("  Sourcetimestamp: ").Append(Sourcetimestamp).Append("\n");
            sb.Append("  Monitored: ").Append(Monitored).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as RealTimeInformation);
        }

        /// <summary>
        /// Returns true if RealTimeInformation instances are equal
        /// </summary>
        /// <param name="other">Instance of RealTimeInformation to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(RealTimeInformation other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Arrivaldatetime == other.Arrivaldatetime ||
                    this.Arrivaldatetime != null &&
                    this.Arrivaldatetime.Equals(other.Arrivaldatetime)
                ) && 
                (
                    this.Duetime == other.Duetime ||
                    this.Duetime != null &&
                    this.Duetime.Equals(other.Duetime)
                ) && 
                (
                    this.Departuredatetime == other.Departuredatetime ||
                    this.Departuredatetime != null &&
                    this.Departuredatetime.Equals(other.Departuredatetime)
                ) && 
                (
                    this.Departureduetime == other.Departureduetime ||
                    this.Departureduetime != null &&
                    this.Departureduetime.Equals(other.Departureduetime)
                ) && 
                (
                    this.Scheduledarrivaldatetime == other.Scheduledarrivaldatetime ||
                    this.Scheduledarrivaldatetime != null &&
                    this.Scheduledarrivaldatetime.Equals(other.Scheduledarrivaldatetime)
                ) && 
                (
                    this.Scheduleddeparturedatetime == other.Scheduleddeparturedatetime ||
                    this.Scheduleddeparturedatetime != null &&
                    this.Scheduleddeparturedatetime.Equals(other.Scheduleddeparturedatetime)
                ) && 
                (
                    this.Destination == other.Destination ||
                    this.Destination != null &&
                    this.Destination.Equals(other.Destination)
                ) && 
                (
                    this.Destinationlocalized == other.Destinationlocalized ||
                    this.Destinationlocalized != null &&
                    this.Destinationlocalized.Equals(other.Destinationlocalized)
                ) && 
                (
                    this.Origin == other.Origin ||
                    this.Origin != null &&
                    this.Origin.Equals(other.Origin)
                ) && 
                (
                    this.Originlocalized == other.Originlocalized ||
                    this.Originlocalized != null &&
                    this.Originlocalized.Equals(other.Originlocalized)
                ) && 
                (
                    this.Direction == other.Direction ||
                    this.Direction != null &&
                    this.Direction.Equals(other.Direction)
                ) && 
                (
                    this._Operator == other._Operator ||
                    this._Operator != null &&
                    this._Operator.Equals(other._Operator)
                ) && 
                (
                    this.Additionalinformation == other.Additionalinformation ||
                    this.Additionalinformation != null &&
                    this.Additionalinformation.Equals(other.Additionalinformation)
                ) && 
                (
                    this.Lowfloorstatus == other.Lowfloorstatus ||
                    this.Lowfloorstatus != null &&
                    this.Lowfloorstatus.Equals(other.Lowfloorstatus)
                ) && 
                (
                    this.Route == other.Route ||
                    this.Route != null &&
                    this.Route.Equals(other.Route)
                ) && 
                (
                    this.Sourcetimestamp == other.Sourcetimestamp ||
                    this.Sourcetimestamp != null &&
                    this.Sourcetimestamp.Equals(other.Sourcetimestamp)
                ) && 
                (
                    this.Monitored == other.Monitored ||
                    this.Monitored != null &&
                    this.Monitored.Equals(other.Monitored)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                if (this.Arrivaldatetime != null)
                    hash = hash * 59 + this.Arrivaldatetime.GetHashCode();
                if (this.Duetime != null)
                    hash = hash * 59 + this.Duetime.GetHashCode();
                if (this.Departuredatetime != null)
                    hash = hash * 59 + this.Departuredatetime.GetHashCode();
                if (this.Departureduetime != null)
                    hash = hash * 59 + this.Departureduetime.GetHashCode();
                if (this.Scheduledarrivaldatetime != null)
                    hash = hash * 59 + this.Scheduledarrivaldatetime.GetHashCode();
                if (this.Scheduleddeparturedatetime != null)
                    hash = hash * 59 + this.Scheduleddeparturedatetime.GetHashCode();
                if (this.Destination != null)
                    hash = hash * 59 + this.Destination.GetHashCode();
                if (this.Destinationlocalized != null)
                    hash = hash * 59 + this.Destinationlocalized.GetHashCode();
                if (this.Origin != null)
                    hash = hash * 59 + this.Origin.GetHashCode();
                if (this.Originlocalized != null)
                    hash = hash * 59 + this.Originlocalized.GetHashCode();
                if (this.Direction != null)
                    hash = hash * 59 + this.Direction.GetHashCode();
                if (this._Operator != null)
                    hash = hash * 59 + this._Operator.GetHashCode();
                if (this.Additionalinformation != null)
                    hash = hash * 59 + this.Additionalinformation.GetHashCode();
                if (this.Lowfloorstatus != null)
                    hash = hash * 59 + this.Lowfloorstatus.GetHashCode();
                if (this.Route != null)
                    hash = hash * 59 + this.Route.GetHashCode();
                if (this.Sourcetimestamp != null)
                    hash = hash * 59 + this.Sourcetimestamp.GetHashCode();
                if (this.Monitored != null)
                    hash = hash * 59 + this.Monitored.GetHashCode();
                return hash;
            }
        }
    }

}

File: sdk/java/src/main/java/io/swagger/client/model/RealTimeInformation.java
/**
 * RTPI REST Web Services API
 * <p>RTPI REST Web Services API (<a href=\"https://goo.gl/s6NTJe\">RRWS API</a>) provides REST interface to retrieve information on real time bus information, timetables and bus stops. </p> <p>Each method available in RRWS API will be invoked using plain HTTP Request (GET method). Depending on parameters provided, result will be returned either in XML or JSON format (if format is not provided then JSON is used by default). Response will contain requested information (see method specification for details) and error status of the request.</p>  <h2>Error Codes</h2> <table> <tr><th>Error Code</th><th>Description</th></tr> <tr><td>0</td><td>Success</td></tr> <tr><td>1</td><td>No Results</td></tr> <tr><td>2</td><td>Missing Parameter</td></tr> <tr><td>3</td><td>Invalid Parameter</td></tr> <tr><td>4</td><td>Sceduled Downtime</td></tr> <tr><td>5</td><td>Unexpected System Error</td></tr> </table> <p>Error code is included in response for each message. For some error codes additional details will be provided in free text format (Error Message field).</p> 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


package io.swagger.client.model;

import java.util.Objects;
import com.google.gson.annotations.SerializedName;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;


/**
 * RealTimeInformation
 */
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2017-06-17T16:51:03.987+01:00")
public class RealTimeInformation   {
  @SerializedName("arrivaldatetime")
  private String arrivaldatetime = null;

  @SerializedName("duetime")
  private String duetime = null;

  @SerializedName("departuredatetime")
  private String departuredatetime = null;

  @SerializedName("departureduetime")
  private String departureduetime = null;

  @SerializedName("scheduledarrivaldatetime")
  private String scheduledarrivaldatetime = null;

  @SerializedName("scheduleddeparturedatetime")
  private String scheduleddeparturedatetime = null;

  @SerializedName("destination")
  private String destination = null;

  @SerializedName("destinationlocalized")
  private String destinationlocalized = null;

  @SerializedName("origin")
  private String origin = null;

  @SerializedName("originlocalized")
  private String originlocalized = null;

  @SerializedName("direction")
  private String direction = null;

  @SerializedName("operator")
  private String operator = null;

  @SerializedName("additionalinformation")
  private String additionalinformation = null;

  @SerializedName("lowfloorstatus")
  private String lowfloorstatus = null;

  @SerializedName("route")
  private String route = null;

  @SerializedName("sourcetimestamp")
  private String sourcetimestamp = null;

  @SerializedName("monitored")
  private String monitored = null;

  public RealTimeInformation arrivaldatetime(String arrivaldatetime) {
    this.arrivaldatetime = arrivaldatetime;
    return this;
  }

   /**
   * Get arrivaldatetime
   * @return arrivaldatetime
  **/
  @ApiModelProperty(example = "17/06/2017 14:40:00", value = "")
  public String getArrivaldatetime() {
    return arrivaldatetime;
  }

  public void setArrivaldatetime(String arrivaldatetime) {
    this.arrivaldatetime = arrivaldatetime;
  }

  public RealTimeInformation duetime(String duetime) {
    this.duetime = duetime;
    return this;
  }

   /**
   * Get duetime
   * @return duetime
  **/
  @ApiModelProperty(example = "Due", value = "")
  public String getDuetime() {
    return duetime;
  }

  public void setDuetime(String duetime) {
    this.duetime = duetime;
  }

  public RealTimeInformation departuredatetime(String departuredatetime) {
    this.departuredatetime = departuredatetime;
    return this;
  }

   /**
   * Get departuredatetime
   * @return departuredatetime
  **/
  @ApiModelProperty(example = "17/06/2017 14:40:00", value = "")
  public String getDeparturedatetime() {
    return departuredatetime;
  }

  public void setDeparturedatetime(String departuredatetime) {
    this.departuredatetime = departuredatetime;
  }

  public RealTimeInformation departureduetime(String departureduetime) {
    this.departureduetime = departureduetime;
    return this;
  }

   /**
   * Get departureduetime
   * @return departureduetime
  **/
  @ApiModelProperty(example = "Due", value = "")
  public String getDepartureduetime() {
    return departureduetime;
  }

  public void setDepartureduetime(String departureduetime) {
    this.departureduetime = departureduetime;
  }

  public RealTimeInformation scheduledarrivaldatetime(String scheduledarrivaldatetime) {
    this.scheduledarrivaldatetime = scheduledarrivaldatetime;
    return this;
  }

   /**
   * Get scheduledarrivaldatetime
   * @return scheduledarrivaldatetime
  **/
  @ApiModelProperty(example = "17/06/2017 14:40:00", value = "")
  public String getScheduledarrivaldatetime() {
    return scheduledarrivaldatetime;
  }

  public void setScheduledarrivaldatetime(String scheduledarrivaldatetime) {
    this.scheduledarrivaldatetime = scheduledarrivaldatetime;
  }

  public RealTimeInformation scheduleddeparturedatetime(String scheduleddeparturedatetime) {
    this.scheduleddeparturedatetime = scheduleddeparturedatetime;
    return this;
  }

   /**
   * Get scheduleddeparturedatetime
   * @return scheduleddeparturedatetime
  **/
  @ApiModelProperty(example = "17/06/2017 14:40:00", value = "")
  public String getScheduleddeparturedatetime() {
    return scheduleddeparturedatetime;
  }

  public void setScheduleddeparturedatetime(String scheduleddeparturedatetime) {
    this.scheduleddeparturedatetime = scheduleddeparturedatetime;
  }

  public RealTimeInformation destination(String destination) {
    this.destination = destination;
    return this;
  }

   /**
   * Get destination
   * @return destination
  **/
  @ApiModelProperty(example = "Shanard Road", value = "")
  public String getDestination() {
    return destination;
  }

  public void setDestination(String destination) {
    this.destination = destination;
  }

  public RealTimeInformation destinationlocalized(String destinationlocalized) {
    this.destinationlocalized = destinationlocalized;
    return this;
  }

   /**
   * Get destinationlocalized
   * @return destinationlocalized
  **/
  @ApiModelProperty(example = "null", value = "")
  public String getDestinationlocalized() {
    return destinationlocalized;
  }

  public void setDestinationlocalized(String destinationlocalized) {
    this.destinationlocalized = destinationlocalized;
  }

  public RealTimeInformation origin(String origin) {
    this.origin = origin;
    return this;
  }

   /**
   * Get origin
   * @return origin
  **/
  @ApiModelProperty(example = "Park Avenue", value = "")
  public String getOrigin() {
    return origin;
  }

  public void setOrigin(String origin) {
    this.origin = origin;
  }

  public RealTimeInformation originlocalized(String originlocalized) {
    this.originlocalized = originlocalized;
    return this;
  }

   /**
   * Get originlocalized
   * @return originlocalized
  **/
  @ApiModelProperty(example = "Park Avenue", value = "")
  public String getOriginlocalized() {
    return originlocalized;
  }

  public void setOriginlocalized(String originlocalized) {
    this.originlocalized = originlocalized;
  }

  public RealTimeInformation direction(String direction) {
    this.direction = direction;
    return this;
  }

   /**
   * Get direction
   * @return direction
  **/
  @ApiModelProperty(example = "Inbound", value = "")
  public String getDirection() {
    return direction;
  }

  public void setDirection(String direction) {
    this.direction = direction;
  }

  public RealTimeInformation operator(String operator) {
    this.operator = operator;
    return this;
  }

   /**
   * Get operator
   * @return operator
  **/
  @ApiModelProperty(example = "bac", value = "")
  public String getOperator() {
    return operator;
  }

  public void setOperator(String operator) {
    this.operator = operator;
  }

  public RealTimeInformation additionalinformation(String additionalinformation) {
    this.additionalinformation = additionalinformation;
    return this;
  }

   /**
   * Get additionalinformation
   * @return additionalinformation
  **/
  @ApiModelProperty(example = "null", value = "")
  public String getAdditionalinformation() {
    return additionalinformation;
  }

  public void setAdditionalinformation(String additionalinformation) {
    this.additionalinformation = additionalinformation;
  }

  public RealTimeInformation lowfloorstatus(String lowfloorstatus) {
    this.lowfloorstatus = lowfloorstatus;
    return this;
  }

   /**
   * Get lowfloorstatus
   * @return lowfloorstatus
  **/
  @ApiModelProperty(example = "no", value = "")
  public String getLowfloorstatus() {
    return lowfloorstatus;
  }

  public void setLowfloorstatus(String lowfloorstatus) {
    this.lowfloorstatus = lowfloorstatus;
  }

  public RealTimeInformation route(String route) {
    this.route = route;
    return this;
  }

   /**
   * Get route
   * @return route
  **/
  @ApiModelProperty(example = "1", value = "")
  public String getRoute() {
    return route;
  }

  public void setRoute(String route) {
    this.route = route;
  }

  public RealTimeInformation sourcetimestamp(String sourcetimestamp) {
    this.sourcetimestamp = sourcetimestamp;
    return this;
  }

   /**
   * Get sourcetimestamp
   * @return sourcetimestamp
  **/
  @ApiModelProperty(example = "17/06/2017 13:40:16", value = "")
  public String getSourcetimestamp() {
    return sourcetimestamp;
  }

  public void setSourcetimestamp(String sourcetimestamp) {
    this.sourcetimestamp = sourcetimestamp;
  }

  public RealTimeInformation monitored(String monitored) {
    this.monitored = monitored;
    return this;
  }

   /**
   * Get monitored
   * @return monitored
  **/
  @ApiModelProperty(example = "true", value = "")
  public String getMonitored() {
    return monitored;
  }

  public void setMonitored(String monitored) {
    this.monitored = monitored;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    RealTimeInformation realTimeInformation = (RealTimeInformation) o;
    return Objects.equals(this.arrivaldatetime, realTimeInformation.arrivaldatetime) &&
        Objects.equals(this.duetime, realTimeInformation.duetime) &&
        Objects.equals(this.departuredatetime, realTimeInformation.departuredatetime) &&
        Objects.equals(this.departureduetime, realTimeInformation.departureduetime) &&
        Objects.equals(this.scheduledarrivaldatetime, realTimeInformation.scheduledarrivaldatetime) &&
        Objects.equals(this.scheduleddeparturedatetime, realTimeInformation.scheduleddeparturedatetime) &&
        Objects.equals(this.destination, realTimeInformation.destination) &&
        Objects.equals(this.destinationlocalized, realTimeInformation.destinationlocalized) &&
        Objects.equals(this.origin, realTimeInformation.origin) &&
        Objects.equals(this.originlocalized, realTimeInformation.originlocalized) &&
        Objects.equals(this.direction, realTimeInformation.direction) &&
        Objects.equals(this.operator, realTimeInformation.operator) &&
        Objects.equals(this.additionalinformation, realTimeInformation.additionalinformation) &&
        Objects.equals(this.lowfloorstatus, realTimeInformation.lowfloorstatus) &&
        Objects.equals(this.route, realTimeInformation.route) &&
        Objects.equals(this.sourcetimestamp, realTimeInformation.sourcetimestamp) &&
        Objects.equals(this.monitored, realTimeInformation.monitored);
  }

  @Override
  public int hashCode() {
    return Objects.hash(arrivaldatetime, duetime, departuredatetime, departureduetime, scheduledarrivaldatetime, scheduleddeparturedatetime, destination, destinationlocalized, origin, originlocalized, direction, operator, additionalinformation, lowfloorstatus, route, sourcetimestamp, monitored);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class RealTimeInformation {\n");
    
    sb.append("    arrivaldatetime: ").append(toIndentedString(arrivaldatetime)).append("\n");
    sb.append("    duetime: ").append(toIndentedString(duetime)).append("\n");
    sb.append("    departuredatetime: ").append(toIndentedString(departuredatetime)).append("\n");
    sb.append("    departureduetime: ").append(toIndentedString(departureduetime)).append("\n");
    sb.append("    scheduledarrivaldatetime: ").append(toIndentedString(scheduledarrivaldatetime)).append("\n");
    sb.append("    scheduleddeparturedatetime: ").append(toIndentedString(scheduleddeparturedatetime)).append("\n");
    sb.append("    destination: ").append(toIndentedString(destination)).append("\n");
    sb.append("    destinationlocalized: ").append(toIndentedString(destinationlocalized)).append("\n");
    sb.append("    origin: ").append(toIndentedString(origin)).append("\n");
    sb.append("    originlocalized: ").append(toIndentedString(originlocalized)).append("\n");
    sb.append("    direction: ").append(toIndentedString(direction)).append("\n");
    sb.append("    operator: ").append(toIndentedString(operator)).append("\n");
    sb.append("    additionalinformation: ").append(toIndentedString(additionalinformation)).append("\n");
    sb.append("    lowfloorstatus: ").append(toIndentedString(lowfloorstatus)).append("\n");
    sb.append("    route: ").append(toIndentedString(route)).append("\n");
    sb.append("    sourcetimestamp: ").append(toIndentedString(sourcetimestamp)).append("\n");
    sb.append("    monitored: ").append(toIndentedString(monitored)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

